### PR TITLE
feat(resty) disable IPv6 resolution by default and introduce an --ipv6 flag

### PR DESCRIPTION
**Pull request by thibaultcha**
Handy when scripting with cosockets and if we don't want those to
attempt a connection through IPv6.

### CODE REVIEW

**Comment by thibaultcha** [comment]
Rebased on `master` to solve the conflicts.

**Comment by thibaultcha** [comment]
Update: as discussed previously: needs to be re-worked to use `ipv6=off` by default, and implement an `--ipv6` enabler flag.

**Comment by mikz** [comment]
Opt in for ipv6 sounds great! Just got bitten by this.


Also, was there some discussion about setting custom dns servers rather than 8.8.8.8 ? Or reading from system /etc/resolv.conf ?

**Comment by agentzh** [comment]
@mikz The `resty` tool already first reads your `/etc/resolv.conf` for the default name servers used. Only when there is nothing in your `/etc/resolv.conf`, it will fall back to `8.8.8.8` and `8.8.4.4`.

**Comment by mikz** [comment]
@agentzh yeah. Edited my question after reading the code. I remembered it wrong.

Anyway, would setting it to something else than 8.8.8.8 and 8.8.4.4 via a command line options be an option? In OpenShift environments there are several entries in /etc/resolv.conf. And some of them don't work properly. Solution to that problem is to check they are working and filter them. 

I could prepare a patch if it would be ok for you. Also including this one and negating the logic.

**Comment by agentzh** [comment]
@mikz Yeah, patches welcome :)

**Comment by thibaultcha** [comment]
@mikz If you allow me, I would really like to complete this patch first. Plus I think it is preferred to introduce separate patches for those 2 changes. :)

**Comment by mikz** [comment]
@thibaultcha sure!  Won't be able to do it for some time anyway.

**Comment by thibaultcha** [comment]
I have added a new commit which reverses the logic of this flag, as discussed.

File: bin/resty [hunk 1]
```diff
@@ -407,6 +412,8 @@ Options:
     --main-include path Include the specified file in the nginx main configuration block
                         (multiple instances are supported).
 
+    --ipv6              Make the nginx resolver lookup both IPv4 and IPv6 addresses
```
Comment (thibaultcha) [inline bin/resty]: not sure if this should be `--ipv6` or `-ipv6`.

Comment (agentzh) [inline bin/resty]: @thibaultcha I think `--resolve-ipv6` is better?

File: t/lib/Test/Resty.pm [hunk 2]
```diff
@@ -124,12 +124,19 @@ sub run_test ($) {
...
-    my $regex = $block->out_like;
-    if (defined $regex) {
+    my $regex;
+    if (defined $block->out_like) {
+        $regex = $block->out_like;
         if (!ref $regex) {
             $regex = qr/$regex/ms;
         }
         like $out, $regex, "$name - stdout like okay";
+    } elsif (defined $block->out_not_like) {
```
Comment (thibaultcha) [inline t/lib/Test/Resty.pm]: missing line jump

**Comment by thibaultcha** [comment]
New commit:
- rename flag to `--resolve-ipv6`
- update the README `--help` section
- fix the lib/Test/Resty.pm code style

**Comment by agentzh** [comment]
@thibaultcha Merged with minor edits in your commit log message. Thanks!

**Pull request closed without merging**